src/raftstore/store/fsm/mod.rs

```diff
@@ -70,11 +70,10 @@ pub struct Store<T, C: 'static> {
     region_ranges: BTreeMap<Key, u64>,
     // the regions with pending snapshots between two mio ticks.
     pending_snapshot_regions: Vec<metapb::Region>,
-    // A marker used to indicate if the peer of a region is going to apply a snapshot
-    // with different range.
-    // It assumes that when a peer is going to accept snapshot, it can never
-    // captch up by normal log replication.
-    pending_cross_snap: HashMap<u64, metapb::RegionEpoch>,
+    // target_region_id -> (source_region_id -> merge_target_epoch)
+    pending_merge_targets: HashMap<u64, HashMap<u64, metapb::RegionEpoch>>,
```

Comment:
em, seem the field is not only for mering but also for snapshot now? I think the name is not suitable.

Reply:
not for snapshot